src/uwtools/strings.py

```diff
@@ -0,0 +1,105 @@
+"""
+Canonical strings used throughout uwtools.
+"""
+
+from dataclasses import dataclass, fields
+from typing import Dict, List
+
+
+@dataclass(frozen=True)
+class FORMAT:
+    """
+    A mapping from config format names to literal strings.
+    """
+
+    # Canonical strings:
+
+    _atparse: str = "atparse"
+    _fieldtable: str = "fieldtable"
+    _ini: str = "ini"
+    _jinja2: str = "jinja2"
+    _nml: str = "nml"
+    _sh: str = "sh"
+    _xml: str = "xml"
+    _yaml: str = "yaml"
+
+    # Variants:
+
+    atparse: str = _atparse
+    bash: str = _sh
+    cfg: str = _ini
+    fieldtable: str = _fieldtable
+    ini: str = _ini
+    jinja2: str = _jinja2
+    nml: str = _nml
+    sh: str = _sh
+    yaml: str = _yaml
+    yml: str = _yaml
+
+    @staticmethod
+    def extensions() -> List[str]:
+        """
+        Returns recognized filename extensions.
+        """
+        return [FORMAT.ini, FORMAT.nml, FORMAT.sh, FORMAT.yaml]
+
+    @staticmethod
+    def formats() -> Dict[str, str]:
+        """
+        Returns the recognized format names.
+        """
+        return {
+            field.name: str(getattr(FORMAT, field.name))
+            for field in fields(FORMAT)
+            if not field.name.startswith("_")
+        }
+
+
+@dataclass(frozen=True)
+class STR:
+    """
+    A lookup map for CLI-related strings.
+    """
+
+    action: str = "action"
+    batch: str = "batch"
+    cfgfile: str = "config_file"
+    chgrescube: str = "chgres_cube"
```

Comment:
This is the only one added. Everything else was copied from the original location.